### PR TITLE
[OpenAI] [Inference] TSP definition updates for `v2024_05_01_preview` service release

### PR DESCRIPTION
**Pull request by jpalvarezl**
## Status

Ready for review

## Open questions (using diff between `04-01` -> `05-01` ) 

- What do we do about `FunctionParameters` and `functionObject`. They are defined but unused AFAICT. [Reference](https://github.com/Azure/azure-rest-api-specs/blob/c8e96d090b5cbfbbf3255e4f152965dad756c4e1/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-05-01-preview/inference.yaml#L3790C9-L3807C13)

## Notes

 - It seems like the base branch change happened automatically and it worked fine

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/vector_search.tsp [hunk 1]
```diff
@@ -53,8 +53,10 @@ model OnYourDataEndpointVectorizationSource
   @doc("Specifies the resource endpoint URL from which embeddings should be retrieved. It should be in the format of https://YOUR_RESOURCE_NAME.openai.azure.com/openai/deployments/YOUR_DEPLOYMENT_NAME/embeddings. The api-version query parameter is not allowed.")
   endpoint: url;
 
+  // `authentication` according to the 05-01-preview spec, only supports api-key and data access token auth.
```
Comment (jpalvarezl) [inline specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/vector_search.tsp:56]: It seems that specifically for vector search we can only use a subset of the possible authentication types. Reference: https://github.com/Azure/azure-rest-api-specs/pull/29042/files#diff-c48fe2c850e2a4161ac3a97642fc486eb977ab775fb8a2c9aa2b7b709213b35dR3353-R3375

File: specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/common_options.tsp [hunk 2]
```diff
@@ -38,4 +38,40 @@ alias OnYourDataCommonChatExtensionConfiguration = {
...
+   */
+  @encodedName("application/json", "max_search_queries")
+  @added(ServiceApiVersions.v2024_05_01_Preview)
+  maxSearchQueries?: int32;
+
+  /** 
+   * If specified as true, the system will allow partial search results to be used and the request fails if all the queries fail. 
+   * If not specified, or specified as false, the request will fail if any search query fails.
+   */
+  @encodedName("application/json", "allow_partial_results")
```
Comment (trrwilson) [inline specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/common_options.tsp:54]: I believe this is `allow_partial_result` (no trailing 's'):
https://github.com/Azure/azure-rest-api-specs/blob/cb1185d9961b7dabe002fdb4c3a28c07d130e47e/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-05-01-preview/inference.yaml#L2697

(I like your name better, but the *I suppose* the encoding needs to match)

File: specification/cognitiveservices/OpenAI.Inference/models/completions/azure_chat_extensions.tsp [hunk 3]
```diff
@@ -94,6 +99,51 @@ model AzureChatExtensionDataSourceResponseCitation {
...
+  @encodedName("application/json", "search_queries")
+  searchQueries: string[];
+
+  /** The index of the data source. */
+  @encodedName("application/json", "data_source_index")
+  dataSourceIndex: int32;
+
+  /** The original search score of the retrieved document. */
+  @encodedName("application/json", "original_search_score")
+  originalSearchScore?: float32;
```
Comment (trrwilson) [inline specification/cognitiveservices/OpenAI.Inference/models/completions/azure_chat_extensions.tsp:117]: The significance is unclear, but the spec specifically uses the "double" format for this and the following rerank score, which I believe implies this should be `float64` (double vs. float precision)

File: specification/cognitiveservices/OpenAI.Inference/examples/2024-05-01-preview/start_generate_image.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+  "operationId": "BeginAzureBatchImageGeneration",
+  "title": "Starts the generation of a batch of images from a text caption",
+  "parameters": {
+    "endpoint": "{endpoint}",
+    "api-version": "2024-04-01-preview",
```
Comment (trrwilson) [inline specification/cognitiveservices/OpenAI.Inference/examples/2024-05-01-preview/start_generate_image.json:6]: already on your radar, but we'll want these to match the 2024-05-01-preview in the folder path

File: specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-05-01-preview/inference.yaml [hunk 5]
```diff
@@ -3365,7 +3365,7 @@ components:
               description: Specifies the resource endpoint URL from which embeddings should be retrieved. It should be in the format of https://YOUR_RESOURCE_NAME.openai.azure.com/openai/deployments/YOUR_DEPLOYMENT_NAME/embeddings. The api-version query parameter is not allowed.
               format: uri
             authentication:
-              oneOf:
+                oneOf:
```
Comment (mssfang) [inline specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-05-01-preview/inference.yaml:3368]:
```suggestion
              oneOf:
```

**Review by mssfang — APPROVED** [review]

File: specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/vector_search.tsp [hunk 6]
```diff
@@ -53,8 +53,10 @@ model OnYourDataEndpointVectorizationSource
...
   @doc("Specifies the authentication options to use when retrieving embeddings from the specified endpoint.")
-  authentication: OnYourDataAuthenticationOptions;
+  @added(ServiceApiVersions.v2024_05_01_Preview)
+  authentication: OnYourDataVectorSearchAuthenticationOptions;
```
Comment (mssfang) [inline specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/vector_search.tsp:59]: Is this a known breaking change?

OnYourDataAuthenticationOptions -> OnYourDataVectorSearchAuthenticationOptions

Comment (jpalvarezl) [reply specification/cognitiveservices/OpenAI.Inference/models/completions/extensions/oyd/vector_search.tsp:59]: There was a changed in `v05_01`narrowing down the supported authentication types:

https://github.com/Azure/azure-rest-api-specs/blob/162fb3af1af3a6e166ab16b8f2684c31a6e6bd0b/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-05-01-preview/inference.yaml#L3353-L3375

File: specification/cognitiveservices/OpenAI.Inference/models/completions/azure_chat_extensions.tsp [hunk 7]
```diff
@@ -94,6 +99,51 @@ model AzureChatExtensionDataSourceResponseCitation {
...
+   * Represents the rationale for filtering the document. If the document does not undergo filtering,
+   * this field will remain unset.
+   */
+  @encodedName("application/json", "filter_reason")
+  filterReason?: AzureChatExtensionRetrieveDocumentFilterReason;
+}
+
+/** The reason for filtering the retrieved document. */
+@added(ServiceApiVersions.v2024_05_01_Preview)
+union AzureChatExtensionRetrieveDocumentFilterReason {
```
Comment (johanste) [inline specification/cognitiveservices/OpenAI.Inference/models/completions/azure_chat_extensions.tsp:133]: Given that this is an output only, is it really an extensible enum? Can't the service guarantee the reason documents get filtered staying the same for a given API version?

**Comment by johanste** [comment]
It looks like the generated examples are not included in the readme.md/tagged? (thus the avocado errors).

File: specification/cognitiveservices/OpenAI.Inference/models/completions/azure_chat_extensions.tsp (hunk 7 above)
Comment (jpalvarezl) [reply specification/cognitiveservices/OpenAI.Inference/models/completions/azure_chat_extensions.tsp:133]: Removing `string` as a variant of this `union`

**Review by johanste — APPROVED** [review]

**Pull request merged**